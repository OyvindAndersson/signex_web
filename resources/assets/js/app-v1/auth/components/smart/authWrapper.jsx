import React from 'react'

const defaults = {
    AuthenticatingComponent: () => null, // dont render anything while authenticating
    FailureComponent: () => null, // dont render anything on failure of the predicate
    wrapperDisplayName: 'AuthWrapper'
  }

export default (args) => {
    const { AuthenticatingComponent, FailureComponent, wrapperDisplayName } = {
        ...defaults,
        ...args
      }

    function wrapComponent(WrappedComponent){
        const displayName = WrappedComponent.displayName || WrappedComponent.name || 'Component'

        class AuthWrapper extends React.Component {
            render(){
                const {isAuthenticated, isAuthenticating} = this.props

                if(isAuthenticated){
                    return <WrappedComponent {...this.props} />
                } else if(isAuthenticating){
                    return <AuthenticatingComponent {...this.props} />
                } else {
                    return <FailureComponent {...this.props} />
                }
            }
        }

        return AuthWrapper
    }

    return wrapComponent
}



