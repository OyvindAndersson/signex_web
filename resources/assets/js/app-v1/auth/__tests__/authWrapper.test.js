import React from 'react'
import { connect } from "react-redux"
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import authWrapper from '../components/container/authWrapper'

describe('AuthWrapper component', () => {
    let MainComponent
    let AuthenticatingComponent
    let FailureComponent
    let WrappedComponent

    beforeEach(() => {
        MainComponent = () => (<div>MainComponent</div>)
        AuthenticatingComponent = () => (<div>AuthenticatingComponent</div>)
        FailureComponent = () => (<div>FailureComponent</div>)
        
        WrappedComponent = connect((state) => ({
            isAuthenticated: state.auth.isAuthenticated,
            isAuthenticating: state.auth.isAuthenticating
        }))(authWrapper({ AuthenticatingComponent, FailureComponent })(MainComponent))
    })

    it('should render main component', () => {
        
        const mockStore = configureMockStore()({
            auth: { isAuthenticated: true, isAuthenticating: false }
        });

        let tree = renderer.create(
            <Provider store={mockStore}>
                <WrappedComponent location={ {state: { from: {pathname: "test"}}}} />
            </Provider>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })

    it('should render authenticating component', () => {

        const mockStore = configureMockStore()({
            auth: { isAuthenticated: false, isAuthenticating: true }
        });

        let tree = renderer.create(
            <Provider store={mockStore}>
                <WrappedComponent location={ {state: { from: {pathname: "test"}}}} />
            </Provider>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })

    it('should render failure component', () => {

        const mockStore = configureMockStore()({
            auth: { isAuthenticated: false, isAuthenticating: false }
        });

        let tree = renderer.create(
            <Provider store={mockStore}>
                <WrappedComponent location={ {state: { from: {pathname: "test"}}}} />
            </Provider>
        ).toJSON()
        
        expect(tree).toMatchSnapshot()
    })
})