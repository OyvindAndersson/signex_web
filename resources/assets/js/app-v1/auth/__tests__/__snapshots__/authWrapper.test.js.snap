// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthWrapper component should render authenticating component 1`] = `
<div>
  AuthenticatingComponent
</div>
`;

exports[`AuthWrapper component should render failure component 1`] = `
<div>
  FailureComponent
</div>
`;

exports[`AuthWrapper component should render main component 1`] = `
<div>
  MainComponent
</div>
`;
